core/jobs/src/test/java/com/cognifide/aet/job/common/modifiers/click/ClickModifierTest.java

```diff
@@ -33,91 +31,92 @@
-  }
-
-  public void setupParams(String timeoutValue, String xpathValue) {
-    when(params.get(PARAM_TIMEOUT)).thenReturn(timeoutValue);
-    when(params.get(PARAM_XPATH)).thenReturn(xpathValue);
-  }
-
-  @Test
-  public void setParameters_whenValidParamsAreProvided_thenValidationPassedSuccessfuly() throws ParametersException {
-    setupParams(PARAM_TIMEOUT_VALUE, PARAM_TIMEOUT_VALUE);
-
-    tested.setParameters(params);
-
-    verify(validationResultBuilder, times(0)).addErrorMessage(anyString());
-  }
-
-  @Test
-  public void setParameters_whenInvalidTimeoutParamIsProvided_thenExceptionIsThrown() throws ParametersException {
-    setupParams("xyz", PARAM_XPATH_VALUE);
-
-    tested.setParameters(params);
-
-    verify(validationResultBuilder, times(1)).addErrorMessage(anyString());
-  }
-
-  @Test
-  public void setParameters_whenTooBigTimeoutParamIsProvided_thenExceptionIsThrown() throws ParametersException {
-    setupParams("300000", PARAM_XPATH_VALUE);
-
-    tested.setParameters(params);
-
-    verify(validationResultBuilder, times(1)).addErrorMessage(anyString());
-  }
-
-  @Test
-  public void setParameters_whenNoXpathParamIsProvided_thenExceptionIsThrown() throws ParametersException {
-    setupParams(PARAM_TIMEOUT_VALUE, null);
-
-    tested.setParameters(params);
-
-    verify(validationResultBuilder, times(1)).addErrorMessage(anyString());
-  }
-
-  @Test
-  public void collect_whenValidParamsAreProvided_elementIsClicked() throws ProcessingException, ParametersException {
-    setupParams(PARAM_TIMEOUT_VALUE, PARAM_XPATH_VALUE);
-    when(elementToClick.isDisplayed()).thenReturn(true);
-
-    tested.setParameters(params);
-    tested.collect();
-
-    verify(elementToClick, times(1)).click();
-  }
+    private static final String PARAM_XPATH = "xpath";
+
+    private static final String PARAM_CSS = "css";
+
+    private static final String PARAM_TIMEOUT = "timeout";
+
+    private static final String PARAM_XPATH_VALUE = "//*[@id='toClick']";
+
+    private static final String PARAM_CSS_VALUE = "#hlogo > a";
+
+    private static final String PARAM_TIMEOUT_VALUE = "5";
+
+    @Mock
+    private WebDriver webDriver;
+
+    @InjectMocks
+    private ClickModifier tested;
+
+    @Mock
+    private Map<String, String> params;
+
+    @Mock
+    private WebElement elementToClick;
+
+
+    @Before
+    public void setUp() {
+        when(webDriver.findElement(By.xpath(PARAM_XPATH_VALUE))).thenReturn(elementToClick);
+    }
+
+    public void setupParams(String timeoutValue, String xpathValue, String cssValue) {
+        when(params.get(PARAM_TIMEOUT)).thenReturn(timeoutValue);
+        when(params.get(PARAM_XPATH)).thenReturn(xpathValue);
+        when(params.get(PARAM_CSS)).thenReturn(cssValue);
+    }
+
+    @Test
+    public void setParameters_whenValidXpathParamIsProvided_thenValidationPassedSuccessfuly() throws ParametersException {
+        setupParams(PARAM_TIMEOUT_VALUE, PARAM_TIMEOUT_VALUE, null);
```

Comment:
We are setting `PARAM_TIMEOUT_VALUE` tice in this method.
This probably needs to be fixed. (must have)